src/vmm/src/vm.rs

```diff
@@ -0,0 +1,105 @@
+use crate::vcpu::{self, Vcpu};
+use kvm_bindings::{kvm_pit_config, kvm_userspace_memory_region, KVM_PIT_SPEAKER_DUMMY};
+use kvm_ioctls::{Kvm, VmFd};
+use std::sync::Arc;
+use std::thread;
+use vm_device::device_manager::IoManager;
+use vmm_sys_util::eventfd::EventFd;
+
+/// A KVM specific implementation of a Virtual Machine.
+///
+/// Provides abstractions for working with a VM. Once a generic Vm trait will be available,
+/// this type will become on of the concrete implementations.
+pub struct KvmVm {
+    fd: VmFd,
+    pub(crate) vcpus: Vec<Vcpu>,
+}
+
+#[derive(Debug)]
+pub enum Error {
+    /// Failed to create a VM.
+    CreateVm(kvm_ioctls::Error),
+    /// Failed to setup the user memory regions.
+    SetupMemoryRegion(kvm_ioctls::Error),
+    /// Failed to setup the interrupt controller.
+    SetupInterruptController(kvm_ioctls::Error),
+    /// Failed to create the vcpu.
+    CreateVcpu(vcpu::Error),
+    /// Failed to register IRQ event.
+    RegisterIrqEvent(kvm_ioctls::Error),
+}
+
+// TODO: Implement std::error::Error for Error.
+
+/// Dedicated [`Result`](https://doc.rust-lang.org/std/result/) type.
+pub type Result<T> = std::result::Result<T, Error>;
+
+impl KvmVm {
+    /// Create a new `KvmVm`.
+    pub fn new(kvm: &Kvm) -> Result<Self> {
+        let fd = kvm.create_vm().map_err(Error::CreateVm)?;
+        Ok(KvmVm {
+            fd,
+            vcpus: Vec::new(),
+        })
+    }
+    /// Set the user memory region.
```

Comment:
nit: add newline

Reply:
I kinda changed my mind and now the comment below is not accurate anymore, so I will just remove it :smile: